app/ui/src/main/java/com/fsck/k9/activity/setup/AccountSetupBasics.java

```diff
@@ -369,6 +376,21 @@ private void onNext() {
     @Override
     public void onActivityResult(int requestCode, int resultCode, Intent data) {
         if (resultCode == RESULT_OK) {
+            if (requestCode == CHOOSE_ACCOUNT) {
+                final String accountName = data.getStringExtra(AccountManager.KEY_ACCOUNT_NAME);
+                new AndroidAccountOAuth2TokenStore(this.getApplicationContext()).authorizeApi(accountName, this, new OAuth2TokenProvider.OAuth2TokenProviderAuthCallback() {
+                    @Override
+                    public void success() {
+                        onAccountManagerSetup(accountName);
+                    }
+
+                    @Override
+                    public void failure(AuthorizationException e) {
+                        Timber.e(e);
```

Comment:
There is no error recovery currently, if the user declines, they must go through manual setup or try again clicking `XOAUTH` button...